servicetalk-grpc-protoc/build.gradle

```diff
@@ -27,6 +27,10 @@ apply plugin: "com.google.protobuf"
 
 dependencies {
   implementation project(":servicetalk-annotations")
+
+  // runtimeOnly is enough for this module in isolation, but we want transitive modules to include at compile time too.
+  api project(":servicetalk-data-protobuf")
```

Comment:
not sure if there is a better way to represent this type of dependency. it is somewhat unique bcz we are doing code generation. basically this module doesn't expose it in its API, it doesn't need it to compile this module, however we don't want modules that depend upon `servicetalk-grpc-protoc` to explicitly include this dependency. note that `implementation` doesn't do the trick:

```
/Users/scott/git/servicetalk/servicetalk-examples/grpc/compression/build/generated/sources/proto/main/java/io/grpc/examples/compression/Greeter.java:51: error: package ProtobufSerializerCache does not exist
    private static final MethodDescriptor<HelloRequest, HelloReply> BlockingSayHelloRpc_MD = MethodDescriptors.newMethodDescriptor("/compression.Greeter/SayHello", "sayHello", false, true, HelloRequest.class, "+proto", ProtobufSerializerCache.INSTANCE.serializerDeserializer(HelloRequest.parser()), HelloRequest::getSerializedSize, false, true, HelloReply.class, "+proto", ProtobufSerializerCache.INSTANCE.serializerDeserializer(HelloReply.parser()), HelloReply::getSerializedSize);
```